random-fu/src/Data/Random/Distribution/Uniform.hs

```diff
@@ -120,13 +84,17 @@ boundedEnumStdUniformCDF = enumUniformCDF minBound maxBound
 -- |Compute a uniform random 'Float' value in the range [0,1)
 floatStdUniform :: RVarT m Float
 floatStdUniform = do
-    x <- getRandomWord32
-    return (word32ToFloat x)
+    x <- uniformRangeRVarT (0, 1)
+    -- exclude 1. TODO: come up with something smarter
```

Comment:
This is irksome as it will happen so infrequently but I can't think of a better solution right now.

Reply:
That was one of the things I mentioned on gitter that you can help with ;) I haven't dealt much with floating point number generation much in the past and I don't really feel like diving deep into this topic now

Reply:
Yes I know and I can't think of a way round it - you could try and be clever and pick the smallest FP below 1 I suppose - I feel a bit uncomfortable about that but maybe I shouldn't - I will ping knowledgeable folks and see what they think

Reply:
I don't think my suggestion is a good one. Here's what happens if we change the range: `x * l + (1 -x) * h`. So we would just be replacing a conditional with some arithmetic - better would be to use `uniformDouble01M` (but still do the test).

Reply:
I meant `uniformFloat01M` for this obviously.